pages/src/hooks/useCreateRedirect.ts

```diff
@@ -0,0 +1,58 @@
+'use client'
+
+import { toast, useConfig } from '@payloadcms/ui'
+import { usePluginTranslation } from '../utils/usePluginTranslations.js'
+
+export const useCreateRedirect = () => {
+  const { t } = usePluginTranslation()
+  const {
+    config: {
+      routes: { api },
+      serverURL,
+      collections,
+    },
+  } = useConfig()
+
+  const createRedirect = async (sourcePath: string, destinationPath: string) => {
+    // Find redirects collection slug from config
+    // Look for collection with redirects: true in custom config
+    const redirectsCollection =
+      collections?.find((col) => (col as any).custom?.redirects)?.slug || 'redirects'
```

Comment:
The `custom` field does not exist in the `ClientCollectionConfig` object. I guess that's why you added the `as any`, which does not solve the problem 😄 please avoid this

I think the solution here is to find the redirectsCollection slug in the `SlugField.tsx` (which is a server component), which has access to `CollectionConfig`, which includes the custom field, and then pass this slug down to this hook.